### DISCUSSION TITLE
Using the ask-tell + define-and-run interface with multiple estimators

### DISCUSSION THREAD

**Discussion by gareth-brown-86** [optuna/optuna#5512] [Q&A] [answered]
I am trying to write an interface in sktime to use optuna for the hyperparameter tuning. 
I am using the  ask-tell + define-and-run (here)[https://optuna.readthedocs.io/en/stable/tutorial/20_recipes/009_ask_and_tell.html#define-and-run] and it works really well for a single estimator.
Now I want to extend it to multiple estimator, each that has its own parameters.

In the basic concepts for optuna, there is an example how to create an objective over SVR and RandomForestRegressor.
Is it possible to do this by passing a distribution/distributions to the ask/tell interface?
Here is the distributions that are defined in the example:

```python
from optuna.distributions import FloatDistribution, IntDistribution
param_grid = [
    {
        estimator: sklearn.svm.SVR,
        estimator__C: FloatDistribution( 1e-10, 1e10, log=True)
    },
    {estimator: sklearn.ensemble.RandomForestRegressor,
        estimator__max_depth: IntDistribution('sqrt', 'log2')
    }
]
```

**Comment by HideakiImamura** [comment, marked as answer]
It's OK in Optuna to sampler different parameter set in different trials. The followings are example.

```python
n_trials = 100
for _ in range(n_trials):
    trial = study.ask()
    
    if SOME_CONDITION:
        estimator_C = trial.suggest_float("estimator_C", 1e-10, 1e10, log=True)
        estimator = sklearn.svm.SVR(...)
    else:
        estimator__max_depth = trial.suggest_categorical("estimator__max_depth", ['sqrt', 'log2'])
        estimator = sklearn.ensemble.RandomForestRegressor(...)
    
    acc = score = sklearn.model_selection.cross_val_score(estimator, ...)
    study.tell(trial, acc)

```
